File: src/main/java/com/skyspacefivefiftyfive/service/AgencyService.java
package com.skyspacefivefiftyfive.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspacefivefiftyfive.domain.Agency;
import com.skyspacefivefiftyfive.dto.AgencyDTO;
import com.skyspacefivefiftyfive.dto.AgencySearchDTO;
import com.skyspacefivefiftyfive.dto.AgencyPageDTO;
import com.skyspacefivefiftyfive.dto.AgencyConvertCriteriaDTO;
import com.skyspacefivefiftyfive.service.GenericService;
import com.skyspacefivefiftyfive.dto.common.RequestDTO;
import com.skyspacefivefiftyfive.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface AgencyService extends GenericService<Agency, Integer> {

	List<Agency> findAll();

	ResultDTO addAgency(AgencyDTO agencyDTO, RequestDTO requestDTO);

	ResultDTO updateAgency(AgencyDTO agencyDTO, RequestDTO requestDTO);

    Page<Agency> getAllAgencys(Pageable pageable);

    Page<Agency> getAllAgencys(Specification<Agency> spec, Pageable pageable);

	ResponseEntity<AgencyPageDTO> getAgencys(AgencySearchDTO agencySearchDTO);
	
	List<AgencyDTO> convertAgencysToAgencyDTOs(List<Agency> agencys, AgencyConvertCriteriaDTO convertCriteria);

	AgencyDTO getAgencyDTOById(Integer agencyId);







}

File: skyspacefivefiftyfive_ui/src/services/AgencyService.js
import http from "../http-common"; 

class AgencyService {
  getAllAgencys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/agency/agencys`, searchDTO);
  }

  get(agencyId) {
    return this.getRequest(`/agency/${agencyId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/agency?field=${matchData}`, null);
  }

  addAgency(data) {
    return http.post("/agency/addAgency", data);
  }

  update(data) {
  	return http.post("/agency/updateAgency", data);
  }
  
  uploadImage(data,agencyId) {
  	return http.postForm("/agency/uploadImage/"+agencyId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new AgencyService();
